spec/requests/api/graphql_spec.rb

```diff
@@ -0,0 +1,664 @@
+            },
+            {
+              "uid": "2",
+              "name": "source_a2"
+            },
+            {
+              "uid": "3",
+              "name": "source_b1"
+            },
+            {
+              "uid": "4",
+              "name": "source_b2"
+            },
+            {
+              "uid": "5",
+              "name": "source_b3"
+            }
+          ]
+        }'))
+    end
+
+    it "honors limit parameter" do
+      post(graphql_endpoint, :headers => headers, :params => { "query" => '
+        {
+          sources(limit: 2) {
+            uid
+            name
+          }
+        }' }.to_json)
+
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["data"]).to eq(JSON.parse('
+        {
+          "sources": [
+            {
+              "uid": "1",
+              "name": "source_a1"
+            },
+            {
+              "uid": "2",
+              "name": "source_a2"
+            }
+          ]
+        }'))
+    end
+
+    it "honors offset parameter" do
+      post(graphql_endpoint, :headers => headers, :params => { "query" => '
+        {
+          sources(offset: 1) {
+            uid
+            name
+          }
+        }' }.to_json)
+
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["data"]).to eq(JSON.parse('
+        {
+          "sources": [
+            {
+              "uid": "2",
+              "name": "source_a2"
+            },
+            {
+              "uid": "3",
+              "name": "source_b1"
+            },
+            {
+              "uid": "4",
+              "name": "source_b2"
+            },
+            {
+              "uid": "5",
+              "name": "source_b3"
+            }
+          ]
+        }'))
+    end
+
+    it "honors offset and limit parameter" do
+      post(graphql_endpoint, :headers => headers, :params => { "query" => '
+        {
+          sources(offset: 1, limit: 2) {
+            uid
+            name
+          }
+        }' }.to_json)
+
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["data"]).to eq(JSON.parse('
+        {
+          "sources": [
+            {
+              "uid": "2",
+              "name": "source_a2"
+            },
+            {
+              "uid": "3",
+              "name": "source_b1"
+            }
+          ]
+        }'))
+    end
+
+    it "honors filter parameter" do
+      post(graphql_endpoint, :headers => headers, :params => { "query" => '
+        {
+          sources(filter: { name: { starts_with: "source_b"}}) {
+            name
+          }
+        }' }.to_json)
+
+      expect(response.status).to eq(200)
+      expect(response.parsed_body["data"]).to eq(JSON.parse('
+        {
+          "sources": [
+            {
+              "name": "source_b1"
+            },
+            {
+              "name": "source_b2"
+            },
+            {
+              "name": "source_b3"
+            }
```

Comment:
This feels like it may create sporadic test failures.  That is, unless you are sorting the names somehow, it's possible for them to come back in a different database order, and thus fail the eq matcher.

Reply:
well, PaginatedResponse sorts by id, and the expectation (sorry, couldn't help it 😉) is that the creates by the let!'s above will be done in order, and thus the returned names will be in that order too.  Am I missing some obscure rspec behavior that could bite us later ?